factory/utils.py

```diff
@@ -133,7 +133,10 @@ def __iter__(self):
             if self.next_elements:
                 yield self.next_elements.popleft()
             else:
-                value = next(self.iterator)
+                try:
+                    value = next(self.iterator)
+                except StopIteration:
```

Comment:
Indeed it does crash on 3.7, this fixes it. One thing id like to see is adding 3.7 to tox so this kind of thing is caught sooner. Current error on master:

```
======================================================================
ERROR: test_no_reset (tests.test_utils.ResetableIteratorTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/factory_boy/factory/utils.py", line 136, in __iter__
    value = next(self.iterator)
StopIteration

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/factory_boy/tests/test_utils.py", line 269, in test_no_reset
    self.assertEqual([1, 2, 3], list(i))
RuntimeError: generator raised StopIteration

======================================================================
ERROR: test_reset_after_end (tests.test_utils.ResetableIteratorTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/factory_boy/factory/utils.py", line 136, in __iter__
    value = next(self.iterator)
StopIteration

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/factory_boy/tests/test_utils.py", line 324, in test_reset_after_end
    self.assertRaises(StopIteration, next, iterator)
  File "/usr/local/lib/python3.7/unittest/case.py", line 743, in assertRaises
    return context.handle('assertRaises', args, kwargs)
  File "/usr/local/lib/python3.7/unittest/case.py", line 178, in handle
    callable_obj(*args, **kwargs)
RuntimeError: generator raised StopIteration
```

How to test it:
```
cat << EOF | docker run --rm -i python:3.7 bash
git clone https://github.com/FactoryBoy/factory_boy.git
cd factory_boy/
pip install -r requirements_test.txt 
pip install django==1.11 SQLAlchemy==1.2 mongoengine==0.15 flask flask_sqlalchemy
pip install -e .
apt-get update
yes | apt-get install mongodb
mkdir -p /data/db
mongod > /dev/null &
make test
EOF
```